projects/ccf-rui/src/app/core/store/registration/registration.state.ts

```diff
@@ -141,17 +141,21 @@ export class RegistrationState extends NgxsImmutableDataRepository<RegistrationS
     this.tags = this.injector.get(AnatomicalStructureTagState);
     this.refData = this.injector.get(ReferenceDataState);
 
-    this.globalConfig.config$.pipe(
-      take(1),
-      tap(({ useDownload, register }) => this.ctx.patchState({
-        useRegistrationCallback: !!(!useDownload && register)
-      }))
-    ).subscribe();
-
-    this.globalConfig.getOption('editRegistration').pipe(
-      filterNulls(),
-      tap(reg => this.editRegistration(reg as SpatialEntityJsonLd))
-    ).subscribe();
+    this.refData.getSourceDB().subscribe(db => {
```

Comment:
I think this may cause the database to be loaded twice. Instead listen to the refData state (or part of it) and then subscribe to that.